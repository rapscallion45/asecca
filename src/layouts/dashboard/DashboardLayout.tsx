import { FC, ReactNode } from 'react';
import { styled } from '@mui/material/styles';
import NavBar from '@/components/NavBar/NavBar';

/* background image styling */
const BackgroundStyle = styled('div')(({ theme }) => ({
  display: 'flex',
  position: 'fixed',
  zIndex: 0,
  minHeight: '100vh',
  minWidth: '100%',
  backgroundColor: theme.palette.grey[400],
  // background:
  //   'linear-gradient(0deg, rgba(86,96,156, 0.7), rgba(86,96,156, 0.7)), url(/static/background.webp)',
  // backgroundSize: 'cover',
  // backgroundPositionX: 'center',
  // backgroundPositionY: 'center',
}));

/* root style for the dashboard page content */
const RootStyle = styled('section')({
  display: 'flex',
  position: 'relative',
  minHeight: '100vh',
  minWidth: '100%',
  overflow: 'hidden',
  paddingTop: '100px',
});

interface DashboardLayoutProps {
  children?: ReactNode;
}

/* Base Dashboard Page Layout  */
/* =========================== */
const DashboardLayout: FC<DashboardLayoutProps> = (props) => {
  const { children } = props;

  return (
    <main>
      <BackgroundStyle id="background" />
      <NavBar showLogin />
      <RootStyle id="page-content">{children}</RootStyle>
    </main>
  );
};

export default DashboardLayout;
