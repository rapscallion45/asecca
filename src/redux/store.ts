import { configureStore, Action } from '@reduxjs/toolkit';
import { ThunkAction } from 'redux-thunk';
import logger from 'redux-logger';
import userPermissionReducer from './slices/userPermissionSlice';
import costsConfigReducer from './slices/costsConfigSlice';
import notificationsReducer from './slices/notificationsSlice';
import themeReducer from './slices/themeSlice';
import { collectionsKanbanSlice } from './slices/kanbanSlice';

/**
 * Configuration of the redux store to manage the global application state
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @see See [more info on Redux Store object](https://redux.js.org/api/store)
 *
 * @constant
 * @type {ToolkitStore<Object>}
 */
const store = configureStore({
  reducer: {
    userPermission: userPermissionReducer,
    costsConfig: costsConfigReducer,
    notifications: notificationsReducer,
    theme: themeReducer,
    collectionsKanban: collectionsKanbanSlice.reducer,
  },
  // @ts-ignore
  middleware:
    process.env.NODE_ENV === 'development'
      ? (getDefaultMiddleware) => getDefaultMiddleware().concat(logger)
      : (getDefaultMiddleware) => getDefaultMiddleware(),
});

/**
 * Shorthand application store dispatch type helper
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @typedef AppDispatch - application specific store dispatch type
 */
export type AppDispatch = typeof store.dispatch;

/**
 * Shorthand application state type helper
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @typedef AppState - application specific state object shape
 */
export type AppState = ReturnType<typeof store.getState>;

/**
 * Shorthand application Thunk Action type helper
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @typedef AppThunk - application specific Thunk action definition
 */
export type AppThunk = ThunkAction<void, AppState, unknown, Action<string>>;

export default store;
