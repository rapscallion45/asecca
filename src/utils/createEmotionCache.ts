import createCache from '@emotion/cache';

/* create emotion style cache, and set 'prepend' true to allow MUI style overrides */

const createEmotionCache = () => createCache({ key: 'css', prepend: true });

export default createEmotionCache;
