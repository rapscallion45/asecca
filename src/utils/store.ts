/* redux store configuration placeholder */
const useStore = () => {};
export default useStore;
