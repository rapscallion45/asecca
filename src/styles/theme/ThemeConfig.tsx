import { FC, useMemo, ReactNode } from 'react';
import { useSelector } from 'react-redux';
import { CacheProvider, EmotionCache } from '@emotion/react';
import { CssBaseline } from '@mui/material';
import {
  ThemeProvider,
  createTheme,
  Theme,
  ThemeOptions,
} from '@mui/material/styles';
import useMediaQuery from '@mui/material/useMediaQuery';
import { AppState } from '@/redux/store';
import lightThemePalette from './lightTheme/lightThemePalette';
import darkThemePalette from './darkTheme/darkThemePalette';
import typography from './typography';

/**
 * Theme Config Props
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @see See [more info on creating emotion styling engine cache](https://emotion.sh/docs/introduction)
 *
 * @typedef IThemeConfigProps
 * @prop {EmotionCache} emotionCache - emotion styling engine cache
 * @prop {ReactNode} children - component child node
 */
interface IThemeConfigProps {
  emotionCache: EmotionCache;
  children?: ReactNode;
}

/**
 * Theme Config
 *
 * Global application theme definitions and setup
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @component
 * @param {IThemeConfigProps} props - component props
 * @returns {FC} - theme config functional component
 */
const ThemeConfig: FC<IThemeConfigProps> = (props) => {
  const { emotionCache, children } = props;

  /* grab global theme state */
  const { type: themeType } = useSelector((state: AppState) => state.theme);

  /* check if user has dark mode preference */
  const prefersDarkMode = useMediaQuery('(prefers-color-scheme: dark)');

  /* get options and create theme for current state */
  const themeOptions: ThemeOptions = useMemo(
    () => ({
      palette:
        themeType === 'light' && !prefersDarkMode
          ? lightThemePalette
          : darkThemePalette,
      typography,
    }),
    [themeType, prefersDarkMode]
  );
  const theme: Theme = createTheme(themeOptions);

  return (
    <CacheProvider value={emotionCache}>
      <ThemeProvider theme={theme}>
        <CssBaseline />
        {children}
      </ThemeProvider>
    </CacheProvider>
  );
};

export default ThemeConfig;
