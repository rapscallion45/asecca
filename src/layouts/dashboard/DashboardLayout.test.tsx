import React, { FC } from 'react';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import createEmotionCache from '../../utils/createEmotionCache';
import store from '../../redux/store';
import ThemeConfig from '../../styles/theme/ThemeConfig';
import DashboardLayout from './DashboardLayout';

/**
 * test text
 *
 * @since - 0.0.0
 */
const testMessage = 'Test render message';

/**
 * test child component
 *
 * @since - 0.0.0
 */
const TestChild: FC = () => <div>{testMessage}</div>;

/**
 * Dashboard Layout Unit Tests
 *
 * @author - Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since - 0.0.0
 */
describe('Dashboard Layout', () => {
  it('Renders correctly', async () => {
    /** perform snapshot test */
    const tree = renderer
      .create(
        <Provider store={store}>
          <ThemeConfig emotionCache={createEmotionCache()}>
            <DashboardLayout>
              <TestChild />
            </DashboardLayout>
          </ThemeConfig>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Admin Test Panel & NavBar', () => {
    it('Should be rendered', async () => {
      /** Arrange */
      /** Act */
      render(
        <Provider store={store}>
          <ThemeConfig emotionCache={createEmotionCache()}>
            <DashboardLayout>
              <TestChild />
            </DashboardLayout>
          </ThemeConfig>
        </Provider>
      );

      /** Assert - expect admin test panel and navbar rendered */
      expect(screen.getByText('Admin Test Panel')).toBeInTheDocument();
      expect(screen.getByText('Logout')).toBeInTheDocument();
      expect(screen.getByText(testMessage)).toBeInTheDocument();
    });
  });
});
