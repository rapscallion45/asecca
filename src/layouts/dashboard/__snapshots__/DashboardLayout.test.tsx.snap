// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Layout Renders correctly 1`] = `
<main
  className="css-10650aq"
  id="dashboard-page"
>
  <div
    className="css-sks9iv"
    id="background"
  />
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1xzpisq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1tfylnl-MuiToolbar-root"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-68dm4j-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          color="black"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </button>
      <a
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <img
          alt="Asecca logo"
          data-nimg="1"
          decoding="async"
          height={30}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Flogoblack.webp&w=384&q=75"
          srcSet="/_next/image?url=%2Flogoblack.webp&w=256&q=75 1x, /_next/image?url=%2Flogoblack.webp&w=384&q=75 2x"
          style={
            {
              "color": "transparent",
              "marginTop": "5px",
            }
          }
          width={150}
        />
      </a>
      <div
        className="MuiBox-root css-i9gxme"
      />
      <div
        className="MuiStack-root css-d4yjqk-MuiStack-root"
      >
        <label
          className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1vbggne-MuiFormControlLabel-root"
          data-testid="light-dark-btn"
        >
          <span
            className="MuiSwitch-root MuiSwitch-sizeMedium css-end58r-MuiSwitch-root"
          >
            <span
              className="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-z8z5yk-MuiButtonBase-root-MuiSwitch-switchBase"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <input
                className="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                defaultChecked={false}
                onChange={[Function]}
                required={false}
                type="checkbox"
              />
              <span
                className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
              />
            </span>
            <span
              className="MuiSwitch-track css-j4fnnk-MuiSwitch-track"
            />
          </span>
          <span
            className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-1lyi4-MuiTypography-root"
          />
        </label>
      </div>
    </div>
  </header>
  <div
    className="css-1y6sui9"
  >
    <div
      className="MuiDrawer-root MuiDrawer-docked css-7ik032-MuiDrawer-docked"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-sl5t6j-MuiPaper-root-MuiDrawer-paper"
        style={
          {
            "visibility": undefined,
          }
        }
      >
        <div
          className="css-jjtu05"
        >
          <div
            className="css-19u2xht"
            data-simplebar="init"
            timeout={500}
          >
            <div
              className="simplebar-wrapper"
            >
              <div
                className="simplebar-height-auto-observer-wrapper"
              >
                <div
                  className="simplebar-height-auto-observer"
                />
              </div>
              <div
                className="simplebar-mask"
              >
                <div
                  className="simplebar-offset"
                >
                  <div
                    aria-label="scrollable content"
                    className="simplebar-content-wrapper"
                    role="region"
                    tabIndex={0}
                  >
                    <div
                      className="simplebar-content"
                    >
                      <div
                        className="MuiBox-root css-sgvqor"
                      >
                        <a
                          className="MuiBox-root css-oudrid"
                          href="/"
                          onClick={[Function]}
                          onMouseEnter={[Function]}
                          onTouchStart={[Function]}
                        >
                          <img
                            className="MuiBox-root css-1nbic85"
                            src="/logoblack.webp"
                          />
                        </a>
                      </div>
                      <div
                        className="MuiBox-root css-0"
                      >
                        <ul
                          className="MuiList-root css-1mk9mw3-MuiList-root"
                        >
                          <li
                            className="MuiListSubheader-root MuiListSubheader-gutters MuiListSubheader-sticky css-1n80s42-MuiListSubheader-root"
                          >
                            general
                          </li>
                          <a
                            className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-39qg4h-MuiButtonBase-root-MuiListItemButton-root"
                            href="/"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onContextMenu={[Function]}
                            onDragLeave={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            tabIndex={0}
                            target=""
                          >
                            <svg
                              aria-hidden={true}
                              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1veko3i-MuiSvgIcon-root"
                              data-testid="HomeIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                              />
                            </svg>
                            <div
                              className="MuiListItemText-root css-tlelie-MuiListItemText-root"
                            >
                              home
                            </div>
                          </a>
                          <a
                            className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-39qg4h-MuiButtonBase-root-MuiListItemButton-root"
                            href="/dashboard/account"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onContextMenu={[Function]}
                            onDragLeave={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            tabIndex={0}
                            target=""
                          >
                            <svg
                              aria-hidden={true}
                              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1veko3i-MuiSvgIcon-root"
                              data-testid="AccountCircleIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6zm0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20z"
                              />
                            </svg>
                            <div
                              className="MuiListItemText-root css-tlelie-MuiListItemText-root"
                            >
                              my account
                            </div>
                          </a>
                          <li
                            className="MuiListSubheader-root MuiListSubheader-gutters MuiListSubheader-sticky css-1n80s42-MuiListSubheader-root"
                          >
                            Configurators
                          </li>
                          <a
                            className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-39qg4h-MuiButtonBase-root-MuiListItemButton-root"
                            href="/dashboard/costs-config"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onContextMenu={[Function]}
                            onDragLeave={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            tabIndex={0}
                            target=""
                          >
                            <svg
                              aria-hidden={true}
                              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1veko3i-MuiSvgIcon-root"
                              data-testid="MonetizationOnIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1.41 16.09V20h-2.67v-1.93c-1.71-.36-3.16-1.46-3.27-3.4h1.96c.1 1.05.82 1.87 2.65 1.87 1.96 0 2.4-.98 2.4-1.59 0-.83-.44-1.61-2.67-2.14-2.48-.6-4.18-1.62-4.18-3.67 0-1.72 1.39-2.84 3.11-3.21V4h2.67v1.95c1.86.45 2.79 1.86 2.85 3.39H14.3c-.05-1.11-.64-1.87-2.22-1.87-1.5 0-2.4.68-2.4 1.64 0 .84.65 1.39 2.67 1.91s4.18 1.39 4.18 3.91c-.01 1.83-1.38 2.83-3.12 3.16z"
                              />
                            </svg>
                            <div
                              className="MuiListItemText-root css-tlelie-MuiListItemText-root"
                            >
                              Cost Configurator
                            </div>
                          </a>
                          <li
                            className="MuiListSubheader-root MuiListSubheader-gutters MuiListSubheader-sticky css-1n80s42-MuiListSubheader-root"
                          >
                            Tables
                          </li>
                          <a
                            className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-39qg4h-MuiButtonBase-root-MuiListItemButton-root"
                            href="/dashboard/aggrid-page-1"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onContextMenu={[Function]}
                            onDragLeave={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            tabIndex={0}
                            target=""
                          >
                            <svg
                              aria-hidden={true}
                              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1veko3i-MuiSvgIcon-root"
                              data-testid="TableViewIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M19 7H9c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h10c1.1 0 2-.9 2-2V9c0-1.1-.9-2-2-2zm0 2v2H9V9h10zm-6 6v-2h2v2h-2zm2 2v2h-2v-2h2zm-4-2H9v-2h2v2zm6-2h2v2h-2v-2zm-8 4h2v2H9v-2zm8 2v-2h2v2h-2zM6 17H5c-1.1 0-2-.9-2-2V5c0-1.1.9-2 2-2h10c1.1 0 2 .9 2 2v1h-2V5H5v10h1v2z"
                              />
                            </svg>
                            <div
                              className="MuiListItemText-root css-tlelie-MuiListItemText-root"
                            >
                              Table 1
                            </div>
                          </a>
                          <a
                            className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-39qg4h-MuiButtonBase-root-MuiListItemButton-root"
                            href="/dashboard/aggrid-page-2"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onContextMenu={[Function]}
                            onDragLeave={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            tabIndex={0}
                            target=""
                          >
                            <svg
                              aria-hidden={true}
                              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1veko3i-MuiSvgIcon-root"
                              data-testid="TableViewIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M19 7H9c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h10c1.1 0 2-.9 2-2V9c0-1.1-.9-2-2-2zm0 2v2H9V9h10zm-6 6v-2h2v2h-2zm2 2v2h-2v-2h2zm-4-2H9v-2h2v2zm6-2h2v2h-2v-2zm-8 4h2v2H9v-2zm8 2v-2h2v2h-2zM6 17H5c-1.1 0-2-.9-2-2V5c0-1.1.9-2 2-2h10c1.1 0 2 .9 2 2v1h-2V5H5v10h1v2z"
                              />
                            </svg>
                            <div
                              className="MuiListItemText-root css-tlelie-MuiListItemText-root"
                            >
                              Table 2
                            </div>
                          </a>
                          <a
                            className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-39qg4h-MuiButtonBase-root-MuiListItemButton-root"
                            href="/dashboard/aggrid-page-3"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onContextMenu={[Function]}
                            onDragLeave={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            tabIndex={0}
                            target=""
                          >
                            <svg
                              aria-hidden={true}
                              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1veko3i-MuiSvgIcon-root"
                              data-testid="TableViewIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M19 7H9c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h10c1.1 0 2-.9 2-2V9c0-1.1-.9-2-2-2zm0 2v2H9V9h10zm-6 6v-2h2v2h-2zm2 2v2h-2v-2h2zm-4-2H9v-2h2v2zm6-2h2v2h-2v-2zm-8 4h2v2H9v-2zm8 2v-2h2v2h-2zM6 17H5c-1.1 0-2-.9-2-2V5c0-1.1.9-2 2-2h10c1.1 0 2 .9 2 2v1h-2V5H5v10h1v2z"
                              />
                            </svg>
                            <div
                              className="MuiListItemText-root css-tlelie-MuiListItemText-root"
                            >
                              Table 3
                            </div>
                          </a>
                          <a
                            className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-39qg4h-MuiButtonBase-root-MuiListItemButton-root"
                            href="/dashboard/aggrid-page-4"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onContextMenu={[Function]}
                            onDragLeave={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            tabIndex={0}
                            target=""
                          >
                            <svg
                              aria-hidden={true}
                              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1veko3i-MuiSvgIcon-root"
                              data-testid="TableViewIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M19 7H9c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h10c1.1 0 2-.9 2-2V9c0-1.1-.9-2-2-2zm0 2v2H9V9h10zm-6 6v-2h2v2h-2zm2 2v2h-2v-2h2zm-4-2H9v-2h2v2zm6-2h2v2h-2v-2zm-8 4h2v2H9v-2zm8 2v-2h2v2h-2zM6 17H5c-1.1 0-2-.9-2-2V5c0-1.1.9-2 2-2h10c1.1 0 2 .9 2 2v1h-2V5H5v10h1v2z"
                              />
                            </svg>
                            <div
                              className="MuiListItemText-root css-tlelie-MuiListItemText-root"
                            >
                              Table 4
                            </div>
                          </a>
                        </ul>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="simplebar-placeholder"
              />
            </div>
            <div
              className="simplebar-track simplebar-horizontal"
            >
              <div
                className="simplebar-scrollbar"
              />
            </div>
            <div
              className="simplebar-track simplebar-vertical"
            >
              <div
                className="simplebar-scrollbar"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <section
    className="css-100gjh2"
    id="page-content"
  >
    <div
      className="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhmbct-MuiPaper-root-MuiCard-root"
      >
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1xbr41s-MuiTypography-root"
          >
            Admin Test Panel
          </p>
          <div
            className="MuiBox-root css-h5fkc8"
          >
            <div
              className="MuiFormControl-root MuiTextField-root css-j1gnhe-MuiFormControl-root-MuiTextField-root"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1fz07n4-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink={true}
                htmlFor="user-permission-level-select"
                id="user-permission-level-select-label"
              >
                User Permission
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-rs3fbt-MuiInputBase-root-MuiOutlinedInput-root"
                onClick={[Function]}
              >
                <select
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiNativeSelect-select MuiNativeSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-tgau9p-MuiNativeSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                  disabled={false}
                  id="user-permission-level-select"
                  multiple={false}
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  value="Global"
                >
                  <option
                    value="Global"
                  >
                    Global
                  </option>
                  <option
                    value="Customer"
                  >
                    Customer
                  </option>
                  <option
                    value="Project"
                  >
                    Project
                  </option>
                  <option
                    value="Collection"
                  >
                    Collection
                  </option>
                </select>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiNativeSelect-icon MuiNativeSelect-iconOutlined css-1o9beye-MuiSvgIcon-root-MuiNativeSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden={true}
                  className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="css-14lo706"
                  >
                    <span>
                      User Permission
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <p
            className="MuiTypography-root MuiTypography-body1 css-xsmv6n-MuiTypography-root"
          >
            Selection of simulated user permission level
          </p>
        </div>
      </div>
      <div>
        Test render message
      </div>
    </div>
  </section>
</main>
`;
